Assets/MixedRealityToolkit/Utilities/Editor/PackageManifest/PackageManifestUpdater.cs

```diff
@@ -35,11 +36,95 @@ internal static void EnsureMSBuildForUnity()
             if (!File.Exists(manifestPath))
             {
                 Debug.LogError($"Package manifest file ({manifestPath}) could not be found.");
-                return;
+                return null;
+            }
+
+            return manifestPath;
+        }
+
+        /// <summary>
+        /// Reports whether or not the appropriate version of MSBuild for Unity is specified
+        /// in the Unity Package Manager manifest.
+        /// </summary>
+        /// <returns>
+        /// True if an appropriate verson of MS Build for Unity is configured in the manifest, otherwise false.
+        /// </returns>
+        internal static bool IsMSBuildForUnityEnabled()
+        {
+            string manifestPath = GetPackageManifestFilePath();
+            if (string.IsNullOrWhiteSpace(manifestPath))
+            {
+                return false;
+            }
+
+            // Load the manifest file.
+            string manifestFileContents = File.ReadAllText(manifestPath);
+            if (string.IsNullOrWhiteSpace(manifestFileContents))
+            {
+                return false;
+            }
+
+            // Read the package manifest a line at a time.
+            bool msBuildFound = false;
+            bool isAppropriateVersion = false;
+            Version minVersion = Version.Parse(MSBuildPackageVersion);
+            using (FileStream manifestStream = new FileStream(manifestPath, FileMode.Open, FileAccess.Read))
+            {
+                using (StreamReader reader = new StreamReader(manifestStream))
+                {
+                    // Read the manifest file a line at a time.
+                    while (!reader.EndOfStream)
+                    {
+                        string line = reader.ReadLine();
+                        if (line.Contains(MSBuildPackageName))
```

Comment:
I was asking for tests, but FWIW I don't think that necessarily the file parsing parts are necessarily the most interesting - even having something that would take some expected strings (i.e. the manifest content) and run it through this code would probably be sufficient.

Reply:
fair enough!

Reply:
sounds good. i will rework the code to make it more testable without needing to rewrite files repeatedly